pkg/serverless/streamlogs/streamlogs_test.go

```diff
@@ -0,0 +1,121 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package streamlogs_test package is for testing streamlogs package.
+package streamlogs_test
+
+import (
+	"bytes"
+	"context"
+	"testing"
+	"time"
+
+	"github.com/DataDog/datadog-agent/comp/logs/agent/config"
+	"github.com/DataDog/datadog-agent/pkg/logs/diagnostic"
+	"github.com/DataDog/datadog-agent/pkg/logs/message"
+	"github.com/DataDog/datadog-agent/pkg/logs/sources"
+	"github.com/DataDog/datadog-agent/pkg/serverless/streamlogs"
+	"github.com/stretchr/testify/require"
+)
+
+type messageReceiverGetter struct {
+	bufferedMessageReceiver *diagnostic.BufferedMessageReceiver
+}
+
+func (mrg *messageReceiverGetter) GetMessageReceiver() *diagnostic.BufferedMessageReceiver {
+	return mrg.bufferedMessageReceiver
+}
+
+func newTestMessage(msg string) *message.Message {
+	origin := message.NewOrigin(&sources.LogSource{
+		Name: "AWS Logs",
+		Config: &config.LogsConfig{
+			Type:    "unit-test",
+			Service: "test-lambda",
+			Source:  "lambda",
+			Tags:    []string{"tag1:value1", "tag2:value2"},
+		},
+	})
+	m := message.NewMessageFromLambda(
+		[]byte(msg),
+		origin,
+		"INFO",
+		time.Date(
+			2024,
+			time.May,
+			26,
+			12,
+			14,
+			39,
+			0,
+			time.UTC,
+		),
+		"arn:aws:lambda:ap-northeast-1:111111111111:function:test-lambda",
+		"8b185116-a6d3-41d5-baba-1f430b1f849e",
+		0,
+	)
+	return m
+}
+
+func TestRun(t *testing.T) {
+	type handleMessageArgs struct {
+		message *message.Message
+		rendered []byte
+	}
+	tests := []struct {
+		name             string
+		handleMessageArgsList []handleMessageArgs
+		diableStreamLogs bool
+		wantW            string
+	}{
+		{
+			name:             "disable stream-logs",
+			diableStreamLogs: true,
+			wantW:            "",
+		},
+		{
+			name:             "enable stream-logs",
+			handleMessageArgsList: []handleMessageArgs{
+				{
+					message: newTestMessage("log-1"),
+					rendered: []byte("log-1"),
+				},
+				{
+					message: newTestMessage("ログ2"),
+					rendered: []byte("ログ2"),
+				},
+				{
+					// When logs-agent stops, nil message is sent to stream-logs
+					message: nil,
+					rendered: nil,
+				},
```

Comment:
To make sure this panic will not happen.

![2024-05-27_14-13-50](https://github.com/DataDog/datadog-agent/assets/41987730/a81c147a-b631-493c-bdb5-caa5a9cd7665)